src/components/text-input/_text-input.scss

```diff
@@ -90,3 +90,87 @@
     }
   }
 }
+
+@mixin text-input--experimental {
+  .#{$prefix}--text-input {
+    @include reset;
+    @include typescale('zeta');
+    background-color: $field-01;
+    width: 100%;
+    height: rem(40px);
+    padding: 0 $spacing-xl 0 $spacing-md;
+    color: $text-02;
+    border: none;
+    outline: 1px solid $ui-04;
+    outline-offset: -1px;
+  }
+
+  .#{$prefix}--text-input ~ .#{$prefix}--label {
+    @include typescale('omega');
+    padding: 0;
+    font-weight: 400;
+    display: block;
+    padding: $spacing-sm 0 $spacing-3xl 0;
+    margin: 0;
+  }
+
+  .#{$prefix}--text-input:required ~ .#{$prefix}--label::after {
+    content: ' *';
+  }
+
+  .#{$prefix}--text-input::-webkit-input-placeholder {
+    color: $text-03;
+  }
+
+  .#{$prefix}--text-input:hover {
+    color: $text-01;
+    outline: 1px solid $brand-01;
+    outline-offset: -1px;
+  }
+
+  .#{$prefix}--text-input:focus,
+  .#{$prefix}--text-input:active {
+    outline: 2px solid $brand-01;
+    outline-offset: -2px;
+  }
+
+  .#{$prefix}--text-input:disabled {
+    cursor: not-allowed;
+    pointer-events: none;
+    color: $text-03;
+  }
```

Comment:
The disabled styles look almost identical to the regular styles, besides a small change in color for the label. I think this could cause confusion for users...

Reply:
Agreed, I just asked design to post the specs for this component.